modules/programs/vscode/haskell.nix

```diff
@@ -0,0 +1,102 @@
+{ pkgs, config, lib, ... }:
+let
+
+  defaultHieNixSourceParams = {
+    owner = "domenkozar";
+    repo = "hie-nix";
+    rev = "6794005f909600679d0b7894d0e7140985920775";
+    sha256 = "0pc90ns0xcsa6b630d8kkq5zg8yzszbgd7qmnylkqpa0l58zvnpn";
+  };
+  defaultHieNixSource = pkgs.fetchFromGitHub defaultHieNixSourceParams;
+
+  defaultHieNixExe = hie-nix.hies + "/bin/hie-wrapper";
+  defaultHieNixExeText = ''hie-nix.hies + "/bin/hie-wrapper"'';
+
+  inherit (pkgs.vscode-utils) buildVscodeMarketplaceExtension;
+  inherit (lib) types;
+  cfg = config.programs.vscode.haskell;
+
+  hieNixArgs = lib.optionalAttrs (!cfg.hieNixUsePinnedPkgs) {
+    inherit pkgs;
+  };
+  hie-nix = import cfg.hieNixSource hieNixArgs;
+
+in
+{
+  options.programs.vscode.haskell.enable = lib.mkEnableOption "Haskell integration for Visual Studio Code";
+
+  options.programs.vscode.haskell.hieNixSource = lib.mkOption {
+    type = lib.types.path;
+    default = defaultHieNixSource;
```

Comment:
I think there should be no default here to force the user to choose a hie package themselves.